data/tutorials/language/0lg_00_basic_datatypes.md

```diff
@@ -0,0 +1,808 @@
+- : string = "This is really disco!"
+```
+
+The first expression is another anonymous function. It is the _identity_ function, it can be applied to anything, and it returns its argument unchanged. This means that its parameter can be of any type, and its result has the same type. The same code can be applied to data of different types. This is called _polymorphism_.
+
+Remember, the `'a` is a _type variable_, so values of any type can be passed to the function and replaces the type variable. The identity function has the same input and output type, whatever it may be.
+
+The two following expressions show that the identity function can apply to different type parameters:
+
+```ocaml
+# let f = fun x -> x * x;;
+val f : int -> int = <fun>
+
+# f 9;;
+- : int = 81
+```
+
+Defining a function is the same as naming a value, as illustrated in the first expression:
+
+```ocaml
+# let g x = x * x;;
+val g : int -> int = <fun>
+
+# g 9;;
+- : int = 81
+```
+
+Executable OCaml code consists primarily of functions, so it's beneficial to make them as concise and clear as possible. The function `g` is defined here using a shorter, more common, and maybe more intuitive syntax.
+
+In OCaml, functions may terminate without returning the expected type value by throwing an exception (`exn`), which does not appear in its type. There is no way to know if a function may raise an exception without inspecting its code.
+```ocaml
+# raise;;
+- : exn -> 'a' = <fun>
+```
+
+Exceptions are discussed in the [Error Handling](/docs/error-handling) guide.
+
+Functions may have several parameters.
+```ocaml
+# fun a b -> a ^ " " ^ b;;
+- : string -> string -> string = <fun>
+
+# let mean a b = (a + b) / 2;;
+val mean : int -> int -> int = <fun>
+```
+
+Like the product type symbol `*`, the function type symbol `->` is not associative. The following two types are not the same:
+- `(int -> int) -> int` : This function takes a function of type `int -> int` as a parameter and returns an `int` as the result.
+- `int -> (int -> int)` : This function takes an `int` asa  parameter and returns a function of type `int -> int` as the result.
+
+### Unit
+
+Uniquely, the type `unit` has only one value. It is written `()` and pronounced “unit.”
+
+The `unit` type has several uses. Mainly, it serves as a token when a function does not need to be passed data or doesn't have any data to return once it has completed its computation. This happens when functions have side effects such as OS-level I/O. Functions need to be applied to something for their computation to be triggered, and they also must return something. When nothing meaningful can be passed or returned, `()` should be used.
+```ocaml
+# read_line;;
+- : unit -> string = <fun>
+
+# print_endline;;
+- : string -> unit = <fun>
+```
+
+The function `read_line` reads an end-of-line terminated sequence of characters from standard input and returns it as a string. Reading input begins when `()` is passed.
```

Comment:
Should we show an example here? Perhaps what it looks like to pass `()` to `read_line`?

Reply:
Yes, but there is a hicup. It would be nice to have something like this:
```ocaml
 # read_line ();;
foo bar
- : string = "foo bar"
```

However, we have to say “replace `foo bar` by your own text, followed by end of line”